src/store/workbenchReducer.ts

```diff
@@ -1,14 +1,26 @@
 import { PayloadAction } from '@reduxjs/toolkit';
 import { IColumnDesc } from 'lineupjs';
 import { IRow } from 'tdp_core';
-import { EWorkbenchDirection, ISelectedMapping, IWorkbench } from './interfaces';
+import { EWorkbenchDirection, ISelectedMapping, IWorkbench, IOrdinoAppState } from './interfaces';
 
 export const workbenchReducers = {
-  addFirstWorkbench(state, action: PayloadAction<IWorkbench>) {
+  addFirstWorkbench(
+    state,
+    action: PayloadAction<{
+      workbench: IWorkbench;
+      globalQuery: IOrdinoAppState['globalQuery'];
+      appliedQueryFilter: IOrdinoAppState['appliedQueryFilter'];
+    }>,
+  ) {
     state.focusWorkbenchIndex = 0;
-    state.workbenches = [action.payload];
+    state.workbenches = [action.payload.workbench];
+    state.globalQuery = action.payload.globalQuery;
+    state.appliedQueryFilter = action.payload.appliedQueryFilter;
```

Comment:
How or when does `globalQuery` and `appliedQueryFilter` get assigned to the first view (= ranking view) of the first workbench (compare to the function `addWorkbench` below)?